tests/integration/rolling_update/test_rolling_update.py

```diff
@@ -294,3 +295,40 @@ def test_num_peas(config):
             + 1  # compound pod tail
             + 1  # gateway
         )
+
+
+@pytest.mark.repeat(5)
+@pytest.mark.timeout(30)
+def test_distinct_executor_ids(config, docs, mocker, reraise):
```

Comment:
isn't this tested in `test_normal`? I agree we can retest with multithreading, but we can then use the `num_replicas`, `num_shards` logic?

Reply:
Wasn't sure it was working properly, as there were more shards. Wanted to have the minimum nr to assert on